actionview/lib/action_view/lookup_context.rb

```diff
@@ -167,6 +169,22 @@ def build_view_paths(paths)
         end
       end
 
+      # #build_view_paths creates a new PathSet on each invocation. This helper
+      # caches stringish path arguments, enabling template caching for
+      # dynamically added view paths (e.g. added in a request context).
+      def cache_paths(paths)
+        Array(paths).map do |path|
+          if ActionView::Resolver.caching? && (path.is_a?(String) || path.is_a?(Pathname))
+            # Let PathSet transform `path` and cache the result
+            LookupContext.view_path_cache.fetch_or_store(path.to_s) do
+              PathSet.new([path]).paths[0]
+            end
```

Comment:
This feels like an workaround .We are creating a path set just to discard it and get the path returned by the `PathSet`. I feel like the right fix would to do it inside the `PathSet` initializer.

Reply:
I started out doing the fix in `ActionView::PathSet#typecast`, but it broke other tests. I could not figure out how to handle them correctly, so I reasoned `ActionView::LookupContext` was a better place to fix `prepend_view_path` – scoped to view lookups only.

(This `PathSet.new([path]).paths[0]` thing is just to not duplicate the typecasting logic of `ActionView::PathSet`. It could be written differently if you prefer.)

---

Find a new commit below, which moves the fix to `ActionView::PathSet#typecast`. A few other tests needed to be fixed – but my fix feels like a hack. Please let me know how to handle this better.